apps/gallery/test/marionette/lib/fullscreen_view.js

```diff
@@ -0,0 +1,331 @@
+        Fullscreen_View.Selector.effectOptions);
+    },
+
+    /**
+     * @return {Marionette.Element} Container to host the enhance options' tab.
+     */
+    get enhanceOptions() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.enhanceOptions);
+    },
+
+    /**
+     * @return {Marionette.Element} 
+     * Element to swipe to change exposure settings.
+     */
+    get exposureSlider() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.exposureSlider);
+    },
+
+    /**
+     * @return {Marionette.Element} 
+     * Element to click to save changes from editing.
+     */
+    get editSaveButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.editSaveButton);
+    },
+
+    /**
+     * @return {Marionette.Element} Element to apply edits in edit tool.
+     */
+    get editToolApplyButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.editToolApplyButton);
+    },
+
+    /**
+     * @return {Marionette.Element} Element to click to crop an image.
+     */
+    get editCropAspectPortraitButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.editCropAspectPortraitButton);
+    },
+
+    /**
+     * @return {Marionette.Element} Element to click to apply a sepia affect.
+     */
+    get editEffectSepiaButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.editEffectSepiaButton);
+    },
+
+    /**
+     * @return {Marionette.Element} Container element to host fullscreen images.
+     */
+    get fullscreenFrame2() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.fullscreenFrame2);
+    },
+
+    /**
+     * @return {Marionette.Element} Container element to host fullscreen images.
+     */
+    get fullscreenFrame3() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.fullscreenFrame3);
+    },
+
+    /**
+     * @return {Marionette.Element} Done Button to finish crop and pick image.
+     */
+    get cropDoneButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.cropDoneButton);
+    },
+
+    /**
+     * @return {Marionette.Element} edit crop canvas showing crop overlay.
+     */
+    get editCropCanvas() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.editCropCanvas);
+    },
+
+
+
+    /**
+     * @return {Marionette.Element} Element to click for sharing image.
+     */
+    get shareButton() {
+      return this.client.helper.waitForElement(
+        Fullscreen_View.Selector.shareButton);
+    },
+
+    /**
+    * For gallery app open activity, check opened image src
+    * is set with URL
+    */
+    hasSrcImageBlobURL: function(appURL) {
+      var url = 'blob:' + appURL;
+      return this.displayedImage
+                 .getAttribute('src').indexOf(url) > -1;
+    },
+
```

Comment:
actually, I think below is better:
    /**
    * Check referenced image src
    * is set with the URL provided
    */
    hasSrcImageBlobURL: function(appURL, image) {
      var url = 'blob:' + appURL;
      return image.getAttribute('src').indexOf(url) > -1;
    },
and use
// Check if displayed image src is set with blob url
    assert.ok(fullscreenView.hasSrcImageBlobURL(Gallery.ORIGIN,fullscreenView.openActivityImage));